onnxruntime/core/framework/session_state_initializer.cc

```diff
@@ -101,24 +106,24 @@ common::Status SessionStateInitializer::CreatePlan(const std::vector<NodeArg*>&
   return Status::OK();
 }
 
-common::Status SessionStateInitializer::InitializeAndSave(bool enable_memory_pattern,
-                                                          const std::vector<NodeArg*>* implicit_inputs) {
+common::Status SessionStateInitializer::InitializeAndSave(const std::vector<NodeArg*>* implicit_inputs) {
   const auto* exec_plan_ptr = session_state_.GetExecutionPlan();
   ORT_ENFORCE(exec_plan_ptr, "Execution plan was not found in SessionState. CreatePlan must be called first.");
 
   const auto& exec_plan{*exec_plan_ptr};
   const auto& mlvalue_name_idx_map{session_state_.GetMLValueNameIdxMap()};
 
   // lambda to save initialized tensors into SessionState directly
-  auto add_initialized_tensor = [this](int idx, const onnxruntime::MLValue& value) {
-    session_state_.AddInitializedTensor(idx, value);
-  };
-
-  ORT_RETURN_IF_ERROR(SaveInitializedTensors(graph_, enable_memory_pattern, exec_plan, execution_providers_,
-                                             mlvalue_name_idx_map, session_state_.GetMutableWeightsBuffers(),
-                                             add_initialized_tensor, logger_));
-
-  graph_.CleanAllInitializedTensors();  // remove weights from the graph now to save memory
+  const Env& env = Env::Default();
+  ORT_RETURN_IF_ERROR(
+      SaveInitializedTensors(env, graph_loc_, graph_, exec_plan, execution_providers_, mlvalue_name_idx_map,
+                             session_state_.GetMutableWeightsBuffers(),
+                             [this](int idx, const onnxruntime::MLValue& value, const OrtDeleter& d) -> Status {
+                               return session_state_.AddInitializedTensor(idx, value, d);
+                             },
+                             logger_));
+  // remove weights from the graph now to save memory, but it won't save memory if enable_memory_pattern is true
```

Comment:
>but it won't save memory if enable_memory_pattern is true [](start = 55, length = 57)

helpful to add the 'why' to the comment ('it won't save memory because...') so that's captured

however enable_memory_pattern is always true now, so perhaps any explanation of why memory won't be saved should be in a comment in CleanInitializedTensors rather than here.